compute/main.go

```diff
@@ -0,0 +1,596 @@
+					Properties: &armnetwork.SecurityRulePropertiesFormat{
+						SourceAddressPrefix:      to.StringPtr("0.0.0.0/0"),
+						SourcePortRange:          to.StringPtr("*"),
+						DestinationAddressPrefix: to.StringPtr("0.0.0.0/0"),
+						DestinationPortRange:     to.StringPtr("22"),
+						Protocol:                 armnetwork.SecurityRuleProtocolTCP.ToPtr(),
+						Access:                   armnetwork.SecurityRuleAccessAllow.ToPtr(),
+						Priority:                 to.Int32Ptr(100),
+						Description:              to.StringPtr("sample network security group outbound port 22"),
+						Direction:                armnetwork.SecurityRuleDirectionOutbound.ToPtr(),
+					},
+				},
+			},
+		},
+	}
+
+	pollerResponse, err := nsgClient.BeginCreateOrUpdate(ctx, resourceGroupName, nsgName, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	return resp.NetworkSecurityGroup, nil
+}
+
+func deleteNetworkSecurityGroup(ctx context.Context, connection *armcore.Connection, nsgName string) (*http.Response, error) {
+	nsgClient := armnetwork.NewNetworkSecurityGroupsClient(connection, subscriptionId)
+
+	pollerResponse, err := nsgClient.BeginDelete(ctx, resourceGroupName, nsgName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func createPublicIP(ctx context.Context, connection *armcore.Connection, publicIPName string) (*armnetwork.PublicIPAddress, error) {
+	publicIPAddressClient := armnetwork.NewPublicIPAddressesClient(connection, subscriptionId)
+
+	parameters := armnetwork.PublicIPAddress{
+		Resource: armnetwork.Resource{
+			Location: to.StringPtr(location),
+		},
+		Properties: &armnetwork.PublicIPAddressPropertiesFormat{
+			PublicIPAllocationMethod: armnetwork.IPAllocationMethodStatic.ToPtr(), // Static or Dynamic
+		},
+	}
+
+	pollerResponse, err := publicIPAddressClient.BeginCreateOrUpdate(ctx, resourceGroupName, publicIPName, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	return resp.PublicIPAddress, err
+}
+
+func deletePublicIP(ctx context.Context, connection *armcore.Connection, publicIPName string) (*http.Response, error) {
+	publicIPAddressClient := armnetwork.NewPublicIPAddressesClient(connection, subscriptionId)
+
+	pollerResponse, err := publicIPAddressClient.BeginDelete(ctx, resourceGroupName, publicIPName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	return resp, err
+}
+
+func createNetWorkInterface(ctx context.Context, connection *armcore.Connection, subnetID *string, publicIPID *string, networkSecurityGroupID *string) (*armnetwork.NetworkInterface, error) {
```

Comment:
I would like the ID we are passing in to be of type `string` instead of `*string` like in function `createVirtualMachine`
```suggestion
func createNetWorkInterface(ctx context.Context, connection *armcore.Connection, subnetID string, publicIPID string, networkSecurityGroupID string) (*armnetwork.NetworkInterface, error) {
```